content/dotnet-template-azure-iot-edge-module/CSharp/ModuleBackgroundService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+using System.Text;
+
+namespace SampleModule;
+
+internal class ModuleBackgroundService : BackgroundService
+{
+    private int _counter;
+    private ModuleClient? _moduleClient;
+    private readonly ILogger<ModuleBackgroundService> _logger;
+
+    public ModuleBackgroundService(ILogger<ModuleBackgroundService> logger) => _logger = logger;
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        MqttTransportSettings mqttSetting = new(TransportType.Mqtt_Tcp_Only);
+        ITransportSettings[] settings = { mqttSetting };
+
+        // Open a connection to the Edge runtime
+        _moduleClient = await ModuleClient.CreateFromEnvironmentAsync(settings);
+        await _moduleClient.OpenAsync(stoppingToken);
```

Comment:
If the `stoppingToken` is signaled, this will throw an exception of `OperationCanceledException`. Do you want that to crash the process, or shall we have it shut down gracefully?

Reply:
as this will run inside a container, I think it's ok to crash and let docker to restart the container